### PR TITLE
GRASS 8.2 (new formula)

### PR DESCRIPTION
**Pull request by alexm98**
Added a new formula for [GRASS GIS](grass.osgeo.org/) 8.2.0.

### CODE REVIEW

File: Formula/grass.rb [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+
+    system "./configure", "--prefix=#{prefix}", *flags
+    # system "./configure", "--prefix=#{env['HOME']}/Applications", *flags
+
+    system "sed", "-ibak", "s|/Library|~/Library|g", "macosx/Makefile"
+
+    system "make", "-j", Hardware::CPU.cores
+    system "make", "-j", Hardware::CPU.cores, "install"
+
+    post_install
```
Comment (carlocab) [inline Formula/grass.rb:180]:
```suggestion
```
`post_install` should not be called during `install`.

File: Formula/grass.rb [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+      <string>12.3</string>
+      <key>LSHasLocalizedDisplayName</key>
+      <true/>
+        </dict>
+      </plist>
+    EOS
+
+    # File.open("#{grass_macos_app}/Contents/Info.plist", "w") { |file| file.write(config) }
+    File.write("#{grass_macos_app}/Contents/Info.plist", config)
+  end
```
Comment (carlocab) [inline Formula/grass.rb:309]: This is unmaintainable. If the `Makefile` doesn't produce this then we don't want it.

File: Formula/grass.rb [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+      If it is the case that you can change the shebang at the beginning of
+      the script to enforce Python 3 usage.
+
+        \e[32m#!/usr/bin/env python\e[0m
+
+      Should be changed into
+
+        \e[32m#!/usr/bin/env python3\e[0m
+
+    EOS
```
Comment (carlocab) [inline Formula/grass.rb:322]: We should just change any shebangs needed instead of adding it to `caveats`

File: Formula/grass.rb [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+
+    if head_only?
+      s += <<~EOS
+        This build of GRASS has been compiled without the WxPython GUI.
+
+        The command line tools remain fully functional.
+      EOS
+    end
```
Comment (carlocab) [inline Formula/grass.rb:330]:
```suggestion
```
This is dead code

File: Formula/grass.rb [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+
+    s += <<~EOS
+      You may also copy \e[32mGRASS.app\e[0m to \e[32m/Applications\e[0m or \e[32m~/Applications\e[0m:
+
+        \e[32mcp -r `find $(brew --prefix) -name "GRASS*.app"` /Applications/GRASS.app\e[0m
+    EOS
```
Comment (carlocab) [inline Formula/grass.rb:336]:
```suggestion
```
We don't want to recommend this

File: Formula/grass.rb [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+    system bin/"grass#{majmin_ver}", "--version"
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # This test will fail and we won't accept that! For Homebrew/homebrew-core
+    # this will need to be a test that verifies the functionality of the
+    # software. Run the test with `brew test grass`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
```
Comment (carlocab) [inline Formula/grass.rb:350]:
```suggestion
```
The comments aren't necessary.

File: Formula/grass.rb [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+    s += <<~EOS
+      You may also copy \e[32mGRASS.app\e[0m to \e[32m/Applications\e[0m or \e[32m~/Applications\e[0m:
+
+        \e[32mcp -r `find $(brew --prefix) -name "GRASS*.app"` /Applications/GRASS.app\e[0m
+    EOS
+    s
+  end
+
+  test do
+    system bin/"grass#{majmin_ver}", "--version"
```
Comment (carlocab) [inline Formula/grass.rb:341]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/grass.rb [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+
+  resource "python-pillow" do
+    url "https://files.pythonhosted.org/packages/8c/92/2975b464d9926dc667020ed1abfa6276e68c3571dcb77e43347e15ee9eed/Pillow-9.2.0.tar.gz"
+    sha256 "75e636fd3e0fb872693f23ccb8a5ff2cd578801251f3a4f6854c6a5d437d3c04"
+  end
+
+  def majmin_ver
+    ver_split = version.to_s.split(".")
+    ver_split[0] + ver_split[1]
+  end
```
Comment (carlocab) [inline Formula/grass.rb:80]:
```suggestion
```
Use `version.major_minor`

File: Formula/grass.rb [hunk 9]
```diff
@@ -0,0 +1,352 @@
...
+
+  resource "python-termcolor" do
+    url "https://files.pythonhosted.org/packages/b8/99/18ae745be732ad1cdb0cab8b63848fe6f3ba813e1324c6a689182e527083/termcolor2-0.0.3.tar.gz"
+    sha256 "63ad2eaf1801c919cbeca60a62c099b330338740c8cc4422717b236f3c8f98a7"
+  end
+
+  resource "python-pillow" do
+    url "https://files.pythonhosted.org/packages/8c/92/2975b464d9926dc667020ed1abfa6276e68c3571dcb77e43347e15ee9eed/Pillow-9.2.0.tar.gz"
+    sha256 "75e636fd3e0fb872693f23ccb8a5ff2cd578801251f3a4f6854c6a5d437d3c04"
+  end
```
Comment (carlocab) [inline Formula/grass.rb:75]: Replace with
```ruby
depends_on "pillow"
```

File: Formula/grass.rb [hunk 10]
```diff
@@ -0,0 +1,352 @@
...
+
+  resource "python-dateutil" do
+    url "https://files.pythonhosted.org/packages/4c/c4/13b4776ea2d76c115c1d1b84579f3764ee6d57204f6be27119f13a61d0a9/python-dateutil-2.8.2.tar.gz"
+    sha256 "0123cacc1627ae19ddf3c27a5de5bd67ee4586fbdd6440d9748f8abb483d3e86"
+  end
+
+  resource "python-six" do
+    url "https://files.pythonhosted.org/packages/71/39/171f1c67cd00715f190ba0b100d606d440a28c93c7714febeca8b79af85e/six-1.16.0.tar.gz"
+    sha256 "1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926"
+  end
```
Comment (carlocab) [inline Formula/grass.rb:55]: Replace with
```ruby
depends_on "six"
```

File: Formula/grass.rb [hunk 11]
```diff
@@ -0,0 +1,352 @@
...
+      "--with-tiff",
+      "--with-pdal=#{Formula["pdal"].opt_bin}/pdal-config",
+      "--with-tiff-includes=#{Formula["libtiff"].opt_include}",
+      "--with-tiff-libs=#{Formula["libtiff"].opt_lib}",
+      "--with-png",
+      "--with-png-includes=#{Formula["libpng"].opt_include}",
+      "--with-png-libs=#{Formula["libpng"].opt_lib}",
+      "--with-regex",
+      # "--with-regex-includes=#{Formula["regex-opt"].opt_lib}",
+      # "--with-regex-libs=#{Formula["regex-opt"].opt_lib}",
```
Comment (carlocab) [inline Formula/grass.rb:133]:
```suggestion
```

File: Formula/grass.rb [hunk 12]
```diff
@@ -0,0 +1,352 @@
...
+      "--with-mysql",
+      "--with-mysql-includes=#{Formula["mysql"].opt_include}/mysql",
+      "--with-mysql-libs=#{Formula["mysql"].opt_lib}",
+      "--with-postgres",
+      "--with-postgres-includes=#{Formula["postgresql"].opt_include}",
+      "--with-postgres-libs=#{Formula["postgresql"].opt_lib}",
+      # Error: liblas has been disabled because it is not supported upstream!
+      # "--with-liblas",
+      # "--with-liblas-includes=#{Formula["liblas"].opt_include}",
+      # "--with-liblas-libs=#{Formula["liblas"].opt_lib}",
```
Comment (carlocab) [inline Formula/grass.rb:149]:
```suggestion
```

File: Formula/grass.rb [hunk 13]
```diff
@@ -0,0 +1,352 @@
...
+      "--with-opengl=macosx",
+      "--with-opencl",
+      "--with-openmp",
+      "--with-openmp-includes=#{Formula["libomp"].opt_include}",
+      "--with-openmp-libs=#{Formula["gcc"].opt_lib}/gcc/current",
```
Comment (carlocab) [inline Formula/grass.rb:154]: It looks like you're mixing headers from one OpenMP implementation and libraries from another. You shouldn't do this.

File: Formula/grass.rb [hunk 14]
```diff
@@ -0,0 +1,352 @@
...
+    flags << "--with-opengl-framework=#{sdk_frameworkspath}/OpenGL.framework"
+    flags << "--with-opencl-includes=#{sdk_frameworkspath}/OpenCL.framework/Versions/Current/Headers"
+    flags << "--with-opencl-libs=#{sdk_frameworkspath}/OpenCL.framework/Versions/Current/Headers"
+
+    resource("python-six").stage { system "python", *Language::Python.setup_install_args(libexec/"vendor") }
+
+    system "./configure", "--prefix=#{prefix}", *flags
+    # system "./configure", "--prefix=#{env['HOME']}/Applications", *flags
+
+    system "sed", "-ibak", "s|/Library|~/Library|g", "macosx/Makefile"
```
Comment (carlocab) [inline Formula/grass.rb:175]: Use `inreplace` instead of `sed`.

File: Formula/grass.rb [hunk 15]
```diff
@@ -0,0 +1,352 @@
...
+
+    sdk_frameworkspath = "#{MacOS.sdk_path}/System/Library/Frameworks"
+
+    flags << "--with-macosx-sdk=#{MacOS.sdk_path}"
+    flags << "--with-opengl-includes=#{sdk_frameworkspath}/OpenGL.framework/Headers"
+    flags << "--with-opengl-framework=#{sdk_frameworkspath}/OpenGL.framework"
+    flags << "--with-opencl-includes=#{sdk_frameworkspath}/OpenCL.framework/Versions/Current/Headers"
+    flags << "--with-opencl-libs=#{sdk_frameworkspath}/OpenCL.framework/Versions/Current/Headers"
+
+    resource("python-six").stage { system "python", *Language::Python.setup_install_args(libexec/"vendor") }
```
Comment (carlocab) [inline Formula/grass.rb:170]:
```suggestion
```

File: Formula/grass.rb [hunk 16]
```diff
@@ -0,0 +1,352 @@
...
+
+    system "./configure", "--prefix=#{prefix}", *flags
+    # system "./configure", "--prefix=#{env['HOME']}/Applications", *flags
```
Comment (carlocab) [inline Formula/grass.rb:173]:
```suggestion
```

File: Formula/grass.rb [hunk 17]
```diff
@@ -0,0 +1,352 @@
...
+
+    system "sed", "-ibak", "s|/Library|~/Library|g", "macosx/Makefile"
+
+    system "make", "-j", Hardware::CPU.cores
+    system "make", "-j", Hardware::CPU.cores, "install"
```
Comment (carlocab) [inline Formula/grass.rb:178]:
```suggestion
    system "make"
    system "make", "install"
```
`brew` takes care of `-j`.

**Comment by alexm98** [comment]
@carlocab Thank you for your detailed suggestions! I will work them and follow-up with another PR.

File: Formula/grass.rb (hunk 13 above)
Comment (alexm98) [reply Formula/grass.rb:154]: The thing here is that libomp that is available in Homebrew does not contain `GOMP_parallel_start` which is needed for GRASS to build with openMP support.

```
checking whether to use OpenMP... "yes"
checking for location of OpenMP includes... /opt/homebrew/opt/libomp/include
checking omp.h usability... yes
checking omp.h presence... yes
checking for omp.h... yes
checking for location of OpenMP library... /opt/homebrew/opt/libomp/lib
checking for GOMP_parallel_start... no
checking for GOMP_parallel_start in -lgomp... no
configure: error: *** Unable to locate OpenMP library.
```

As I've seen GCC includes contain libgomp. I see that `libgomp` is not available, so I thougt of using it from GCC. This isn't exactly a justification to do this, but I haven't found another way around this. I would appreciate if someone would be kind to point me as to where I could `libgomp` in an openMP implementation via homebrew that I could use as a dependency.

**Comment by SMillerDev** [comment]
Please don't make a new pull request. Just push your changes here.

**Comment by neteler** [comment]
@alexm98 (perhaps also @nilason): is anything missing here? Would be a pity to see this PR lost.

**Comment by nilason** [comment]
> @alexm98 (perhaps also @nilason): is anything missing here? Would be a pity to see this PR lost.

I agree. And by the look of it this is not far from working.

Just a note: `--enable-macosx-app` is [broken](https://github.com/OSGeo/grass/issues/457). *At the moment* there is no way to `make` yourself a proper Mac app using GRASS sources.
The https://github.com/OSGeo/grass/blob/main/macosx/app/Info.plist.in is for example modernised however and may be used to make an "app-shell" (see https://github.com/nilason/grass-conda/blob/df392f3814b9cb0a908d4361a87ab9f7283d6b54/build-grass.sh#L163).

I don't know whether it is expected in brew world to create an `*.app`, just making grass available through command line is pretty straightforward.

**Comment by nilason** [comment]
Other examples to take cues from is https://github.com/macports/macports-ports/blob/master/gis/grass/Portfile (which includes above mentioned app-shell creation) and https://github.com/OSGeo/grass/pull/2661 for config settings.

**Comment by nilason** [comment]
In GRASS source dir: do `./configure --help` for available options. E.g. no need for `ffmpeg`.

Also, why `gcc`, and not `clang`?

**Comment by nilason** [comment]
I suppose you need `libomp` for `--with-openmp`.

**Comment by alexm98** [comment]
> @nilason `gcc` is added only as a build dependency so that we can use the libraries and headers for `openmp`. The problem is that if I only use `libomp`, there will be no `GOMP_parallel_start` that is checked for. We still build using `clang`.

```
"--with-openmp",
"--with-openmp-includes=#{Formula["gcc"].opt_include}",
"--with-openmp-libs=#{Formula["gcc"].opt_lib}/gcc/current",
```
```
checking whether to use OpenMP... "yes"
checking for location of OpenMP includes... /opt/homebrew/opt/gcc/include
checking omp.h usability... yes
checking omp.h presence... yes
checking for omp.h... yes
checking for location of OpenMP library... /opt/homebrew/opt/gcc/lib/gcc/current
checking for GOMP_parallel_start... no
checking for GOMP_parallel_start in -lgomp... yes
```
As opposed to using `libomp`, `./configure` will not find `GOMP_parallel_start`:

```
"--with-openmp-includes=#{Formula["libomp"].opt_include}",
"--with-openmp-libs=#{Formula["libomp"].opt_lib}",
```

```
checking whether to use OpenMP... "yes"
checking for location of OpenMP includes... /opt/homebrew/opt/libomp/include
checking omp.h usability... yes
checking omp.h presence... yes
checking for omp.h... yes
checking for location of OpenMP library... /opt/homebrew/opt/libomp/lib
checking for GOMP_parallel_start... no
checking for GOMP_parallel_start in -lgomp... no
configure: error: *** Unable to locate OpenMP library

**Comment by alexm98** [comment]
Altough, if we look at the [GitHub actions for the last commit ](https://github.com/Homebrew/homebrew-core/actions/runs/3177339875/jobs/5177730947#step:6:353)where `gcc` libraries and headers we're used for `openmp` support, the build fails.

```
  checking whether to use OpenMP... "yes"
  checking for location of OpenMP includes... /opt/homebrew/opt/gcc/include
  checking omp.h usability... no
  checking omp.h presence... no
  checking for omp.h... no
```
This does not happen on my machine with:
```
"--with-openmp",
"--with-openmp-includes=#{Formula["gcc"].opt_include}",
"--with-openmp-libs=#{Formula["gcc"].opt_lib}/gcc/current",
```

**Comment by nilason** [comment]
I have no experience with homebrew, so I cannot tell how things are or should be (just have to make that clear).

I would test with adding
`depends_on "libomp" => :build`
dropping:
`depends_on "gcc" => :build`


keep:
`"--with-openmp"`

but try with dropping:
`"--with-openmp-includes="`
`"--with-openmp-libs=`
in case they will be found by default.

**Comment by alexm98** [comment]
> I have no experience with homebrew, so I cannot tell how things are or should be (just have to make that clear).
> 
> I would test with adding `depends_on "libomp" => :build` dropping: `depends_on "gcc" => :build`
> 
> keep: `"--with-openmp"`
> 
> but try with dropping: `"--with-openmp-includes="` `"--with-openmp-libs=` in case they will be found by default.

This does not seem to work though. Still not able to find `GOMP_parallel_start`

```
checking whether to use OpenMP... "yes"
checking for location of OpenMP includes...
checking omp.h usability... yes
checking omp.h presence... yes
checking for omp.h... yes
checking for location of OpenMP library...
checking for GOMP_parallel_start... no
checking for GOMP_parallel_start in -lgomp... no
configure: error: *** Unable to locate OpenMP library.
```

**Comment by nilason** [comment]
Let us return to OpenMP later. Will it work without setting it at all?

**Comment by nilason** [comment]
Is capital LIB intentional:
`"--with-libs=#{HOMEBREW_PREFIX}/LIB",`
?

**Comment by alexm98** [comment]
It does work without openmp support. Altough as you mentioned, the ```--enable-macosx-app``` does not create a proper macos `.app`.

**Comment by alexm98** [comment]
> Is capital LIB intentional: `"--with-libs=#{HOMEBREW_PREFIX}/LIB",` ?

Totally unintentional.

**Comment by nilason** [comment]
Please correct it and try with openmp again

**Comment by alexm98** [comment]
> Please correct it and try with openmp again

Does not work with `libomp` nor `gcc`. It seems that not finding `GOMP_parallel_start` is still the problem

**Comment by nilason** [comment]
Looks like the library is not found.
Find out what you have installed: `find /opt/homebrew -name "*omp.dylib"` (or wherever you have HOMEBREW_PREFIX).

It is possible it is located in:
`--with-openmp-libs=#{HOMEBREW_PREFIX}/lib/libomp`

config.log might also give you important information on where configure looks for the lib.
(If you don't mind, please share the config.log).

**Comment by alexm98** [comment]
> Looks like the library is not found. Find out what you have installed: `find /opt/homebrew -name "*omp.dylib"` (or wherever you have HOMEBREW_PREFIX).
> 
> It is possible it is located in: `--with-openmp-libs=#{HOMEBREW_PREFIX}/lib/libomp`
> 
> config.log might also give you important information on where configure looks for the lib. (If you don't mind, please share the config.log).

I can use find to find the libraries:

```
/opt/homebrew/lib/libomp.dylib
/opt/homebrew/Cellar/llvm/15.0.3/lib/libomp.dylib
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/libgomp.dylib
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/12/libgomp.dylib
/opt/homebrew/Cellar/libomp/15.0.5/lib/libomp.dylib
/opt/homebrew/Cellar/libomp/15.0.3/lib/libomp.dylib
/opt/homebrew/Cellar/libomp/14.0.6/lib/libomp.dylib
```

At some point in the `config.log` however:

```
configure:14608: checking whether to use OpenMP
configure:14610: result: "yes"
configure:14630: checking for location of OpenMP includes
configure:14637: result:
configure:14656: checking omp.h usability
configure:14656: gcc -c -g -O2  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.sdk    -I/opt/homebrew/include conftest.c >&5
configure:14656: $? = 0
configure:14656: result: yes
configure:14656: checking omp.h presence
configure:14656: gcc -E    -I/opt/homebrew/include conftest.c
configure:14656: $? = 0
configure:14656: result: yes
configure:14656: checking for omp.h
configure:14656: result: yes
configure:14680: checking for location of OpenMP library
configure:14687: result:
configure:14706: checking for GOMP_parallel_start
configure:14706: gcc -o conftest -g -O2  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.sdk   -I/opt/homebrew/include   -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.sdk -Wl,-rpath,${prefix}/GRASS-8.3.app/Contents/MacOS/lib,-rpath,${GISBASE}/lib  -L/opt/homebrew/lib conftest.c    >&5
Undefined symbols for architecture arm64:
  "_GOMP_parallel_start", referenced from:
      _main in conftest-196a3b.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
configure:14706: $? = 1
```

**Comment by nilason** [comment]
The OpenMP issue starts to look like a limitation/bug in Grass configure. Will look into this. Meanwhile, let me suggest you to reopen the PR, if possible; drop for the time being openmp (so far only of relatively limited use in grass, not a must); and optionally manually create `.app` using https://github.com/OSGeo/grass/blob/main/macosx/app/Info.plist.in and https://github.com/OSGeo/grass/blob/main/macosx/app/AppIcon.icns.

**Comment by alexm98** [comment]
> Looks like the library is not found. Find out what you have installed: `find /opt/homebrew -name "*omp.dylib"` (or wherever you have HOMEBREW_PREFIX).
> 
> It is possible it is located in: `--with-openmp-libs=#{HOMEBREW_PREFIX}/lib/libomp`
> 
> config.log might also give you important information on where configure looks for the lib. (If you don't mind, please share the config.log).

Ok so it seems that

```
"--with-openmp",
"--with-openmp-includes=#{HOMEBREW_PREFIX}/opt/libomp/include",
"--with-openmp-libs=#{HOMEBREW_PREFIX}/opt/libomp/lib",
```

And also replacing -lgomp flag to -lomp in configure (since the libraries found in `Formula["libomp"]` are available for usage with `gcc`)

```
inreplace "configure", "-lgomp", "-lomp"
```

This way, I can get GRASS to compile "with `openmp` support.

```
GRASS is now configured for:  aarch64-apple-darwin22.1.0

  Source directory:           /private/tmp/grass-20221128-24155-veqspd/grass-8.2.0
  Build directory:            /private/tmp/grass-20221128-24155-veqspd/grass-8.2.0
  Installation directory:     ${prefix}/GRASS-8.2.app/Contents/MacOS
  Startup script in directory:${exec_prefix}/bin
  C compiler:                 clang -g -O2  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.sdk
  C++ compiler:               clang++  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.sdk
  Building shared libraries:  yes
  OpenGL platform:            Aqua

  MacOSX application:         yes
  MacOSX architectures:
  MacOSX SDK:                 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.sdk

  BLAS support:               yes
  BZIP2 support:              yes
  C++ support:                yes
  Cairo support:              yes
  DWG support:                no
  FFTW support:               yes
  FreeType support:           yes
  GDAL support:               yes
  GEOS support:               yes
  LAPACK support:             no
  Large File support (LFS):   yes
  libLAS support:             no
  MySQL support:              yes
  NetCDF support:             yes
  NLS support:                yes
  ODBC support:               yes
  OGR support:                yes
  OpenCL support:             yes
  OpenGL support:             yes
  OpenMP support:             yes
  PDAL support:               yes
  PNG support:                yes
  POSIX thread support:       yes
  PostgreSQL support:         yes
  Readline support:           yes
  Regex support:              yes
  SQLite support:             yes
  TIFF support:               yes
  X11 support:                yes
  Zstandard support:          yes
```

However, when I open the GRASS application resulted and run a command using `nprocs`, such as `r.neighbors` it will tell me that GRASS is compiled without OpenMP support.

**Comment by nilason** [comment]
you should probably include a [llvm](https://github.com/Homebrew/homebrew-core/blob/master/Formula/llvm%4014.rb) package instead of gcc

**Comment by nilason** [comment]
... probably together with [libomp](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/libomp.rb)

**Comment by alexm98** [comment]
> ... probably together with [libomp](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/libomp.rb)

Even with [this](https://github.com/Homebrew/homebrew-core/actions/runs/3177339875/jobs/5177730947#step:6:353) the tests in GitHub Actions seem to fail...

**Comment by nilason** [comment]
@alexm98 Please try with https://github.com/OSGeo/grass/pull/2692.

**Pull request closed without merging**